genienlp/arguments.py

```diff
@@ -131,42 +139,90 @@ def parse_argv(parser):
+                        help='dump_preds will dump only predictions; dump_embs will dump both prediction and embeddings')
+    parser.add_argument('--bootleg_batch_size', type=int, default=16, help='Batch size used for inference using bootleg')
+    parser.add_argument('--bootleg_prob_threshold', type=float, default=0.5, help='Probability threshold for accepting a candidate for a mention')
+    parser.add_argument('--bootleg_dataset_threads', type=int, default=2, help='Number of threads for parallel processing of dataset in bootleg')
+    parser.add_argument('--bootleg_dataloader_threads', type=int, default=4, help='Number of threads for parallel loading of datasets in bootleg')
+    parser.add_argument('--bootleg_extract_num_workers', type=int, default=4, help='Number of workers for extracing mentions step of bootleg')
+    parser.add_argument('--bootleg_post_process_types', action='store_true', help='Postprocess bootleg types')
+
+    parser.add_argument('--entity_type_agg_method', choices=['average', 'weighted'], default='average', help='Method used to aggregate several type embeddings for a single mention')
+    parser.add_argument("--entity_word_embeds_dropout", default=0.0, type=float, help='Dropout entity word embeddings with this probability when encoding inputs')
+
+    parser.add_argument("--add_types_to_text", default='no', choices=['no', 'insert', 'append'], help='Method for adding types to input text in text-based NER approach')
+    parser.add_argument("--dump_entity_type_pairs", action='store_true', help='Dump entity type pairs')
+    
+    parser.add_argument('--retrieve_method', default='naive', choices=['naive', 'entity-oracle', 'type-oracle', 'bootleg'], type=str,
+                        help='how to retrieve types for entities')
+    
+    parser.add_argument('--lookup_method', default='ngrams', choices=['ngrams', 'smaller_first', 'longer_first'],
+                        help='smaller_first: start from one token and grow into longer spans until a match is found,'
+                             'longer_first: start from the longest span and shrink until a match is found,'
+                             'ngrams: lookup all ngrams in the text and see if there is a match')
+    
+    parser.add_argument('--verbose', action='store_true', help='Print detected types for each token')
+    parser.add_argument('--almond_domains', nargs='+', default=[], help='Domains used for almond dataset; e.g. music, books, ...')
+    parser.add_argument('--features', nargs='+', type=str, default=[],
+                        help='Features that will be extracted for each entity. Order is important')
+    parser.add_argument('--features_size', nargs='+', type=int, default=[], help='Max length of each feature vector. All features are padded up to this length')
+    parser.add_argument('--features_default_val', nargs='+', type=float, default=[], help='Default value used for each feature')
```

Comment:
Please add "ner" or "ned" prefix to all the ner/ned specific arguments (at least those that don't have "bootleg" or "entity" in the name). The name is really confusing otherwise.